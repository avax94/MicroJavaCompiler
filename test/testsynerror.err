Sintaksna greska 2:9 Sintaksna greska
2:7 Uspesan oporavak od greske do ","
Sintaksna greska 2:14 Sintaksna greska
2:12 Uspesan oporavak od greske do ";"
4:20 Nije pronadjen tip N u tabeli simbola
Sintaksna greska 4:22 Sintaksna greska
4:20 Uspesan oporavak od greske kod nasledjivanja klase
Sintaksna greska 5:17 Sintaksna greska
5:12 Uspesan oporavak od greske kod deklarisanje polja klase
Sintaksna greska 15:21 Sintaksna greska
15:19 Uspesan oporavak od greske do ","
Sintaksna greska 15:38 Sintaksna greska
15:35 Uspesan oporavak od greske do ")"
Sintaksna greska 19:33 Sintaksna greska
Uspesan oporavak od greske do ";"
Redefinisan simbol: x u tabeli simbola
Sintaksna greska 19:46 Sintaksna greska
19:44 Uspesan oporavak od greske do "{"
22:8 Ne postoji simbol y u tabeli simbola!
22:12 Ne postoji simbol y u tabeli simbola!
22:12 Sabirci moraju biti tipa int!
Sintaksna greska 22:16 Sintaksna greska
22:14 Uspesan oporavak od greske prilikom izraza
22:4 Izraz je pogresnog tipa!
Sintaksna greska 23:11 Sintaksna greska
23:9 Uspesan oporavak od greske prilikom uslova
Sintaksna greska 27:14 Sintaksna greska
27:12 Uspesan oporavak od greske prilikom izraza
27:8 Izraz je pogresnog tipa!
Sintaksna greska 27:21 Sintaksna greska
27:19 Uspesan oporavak od greske prilikom uslova
Sintaksna greska 27:33 Sintaksna greska
27:32 Uspesan oporavak od greske prilikom izraza
27:26 Izraz je pogresnog tipa!
28:10 Ne postoji simbol y u tabeli simbola!
28:12 Sabirci moraju biti tipa int!
28:10 Nekompatibilni tipovi!
31:10 Nepostojece polje/metoda metoda za varijablu test1
31:10 Simbol noObj nije metoda!
Sintaksna greska 31:19 Sintaksna greska
31:17 Uspesan oporavak od greske prilikon prosledjivanja argumenata funkciji
Sintaksna greska 32:11 Sintaksna greska
32:8 Uspesan oporavak od greske prilikom izraza
34:3 Nedostaje return statement u metodi!
