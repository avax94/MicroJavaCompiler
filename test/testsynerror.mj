program Err
int x, y z, t f;
int niz[];
class test1 extends N A {
  int fld1, fld2 3, fld3, fld4 5;
 {
  void metoda(int x, int y) {
    fld1 = x + y;
  }
  }
}


{
  int funcTest(int x x, int e, int e2 e) {
    return 0;
  }

  int main() test1 obj; int x, y ff; int x, y {
    niz = new int[2];
    obj = new test1;
    x = y + y + - ;
    if(x < > y) {
      x = 2;
    }

    for(x = 0 2; x < > y; x = x -+) {
      x = y-x;
    }

    test1.metoda(1 1, 2);
    niz[x  y] = 2;

   }
}
