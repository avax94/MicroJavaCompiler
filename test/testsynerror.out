=====================SEMANTICKA ANALIZA=========================
Pretraga na (8), nadjeno Fld fld1: int, 1, 1 
Pretraga na (8), nadjeno Var x: int, 1, 1 
Pretraga na (8), nadjeno Var y: int, 2, 1 
Pretraga na (20), nadjeno Var niz: Arr of int, 2, 0 
Pretraga na (21), nadjeno Var obj: Class, 0, 1 
Pretraga na (22), nadjeno Var x: int, 1, 1 
Pretraga na (23), nadjeno Var x: int, 1, 1 
Pretraga na (24), nadjeno Var x: int, 1, 1 
Pretraga na (27), nadjeno Var x: int, 1, 1 
Pretraga na (27), nadjeno Var x: int, 1, 1 
Pretraga na (27), nadjeno Var x: int, 1, 1 
Pretraga na (27), nadjeno Var x: int, 1, 1 
Pretraga na (28), nadjeno Var x: int, 1, 1 
Pretraga na (28), nadjeno Var x: int, 1, 1 
Pretraga na (31), nadjeno Type test1: Class, 0, 0 
Pretraga na (32), nadjeno Var niz: Arr of int, 2, 0 
=====================SINTAKSNA ANALIZA=========================
Pronadjeno: 3 globalnih promenljivih
Pronadjeno: 0 globalnih konstanti
Pronadjeno: 1 globalnih nizova
Pronadjeno: 3 lokalnih promenljivih
Pronadjeno: 2 globalnih funkcija
Pronadjeno: 1 definicija klasa
Pronadjeno: 1 metoda
Pronadjeno: 2 statement blokova
Pronadjeno: 1 poziva funkcija u mainu
Pronadjeno: 3 formalnih argumenata
Pronadjeno: 1 polja klasa
=====================SYMBOL TABLE DUMP=========================
Var $helper: int, 0, 0 
Prog Err: notype, 0, 0 

Var x: int, 1, 0 
Var niz: Arr of int, 2, 0 
Type test1: Class [Fld $vtable: int, 0, 1 Fld fld1: int, 1, 1 Meth metoda: notype, 0, 3 
   Var this: , 0, 1 
   Var x: int, 1, 1 
   Var y: int, 2, 1 
], 0, 0 
Meth funcTest: int, 12, 1 
   Var e: int, 0, 1 

Meth main: int, 20, 0 
   Var obj: Class [], 0, 1 
   Var x: int, 1, 1 

Type int: int, -1, -1 
Type char: char, -1, -1 
Con eol: char, 10, 0 
Con null: Class [], 0, 0 
Meth chr: char, 0, 1 
   Var i: int, 0, 1 

Meth ord: int, 0, 1 
   Var ch: char, 0, 1 

Meth len: int, 0, 1 
   Var arr: Arr of notype, 0, 1 

Type bool: , -1, -1 

